File: solutions/beecrowd/1805/1805.clj
(ns main
  (:require [clojure.string :as str]))

(defn main []
  (loop [line (read-line)]
    (when (seq line)
      (let [[first last] (->> (str/split line #" ")
                              (map #(Integer/parseInt %)))]
        (printf "%d%n" (/ (* (inc (- last first)) (+ first last)) 2)))
      (recur (read-line)))))

(main)

File: solutions/beecrowd/1805/1805.hs
main :: IO ()
main = do
    contents <- getContents
    let results = map processLine (lines contents)
    mapM_ print results
  where
    processLine line = let [first, last] = map read (words line)
                       in (last - first + 1) * (first + last) `div` 2
